pkg/process/config/yaml_config.go

```diff
@@ -29,6 +29,38 @@ func key(pieces ...string) string {
 	return strings.Join(pieces, ".")
 }
 
+func (a *AgentConfig) setupChecks() {
+	if config.Datadog.IsSet(key(ns, "enabled")) {
+		// A string indicate the enabled state of the Agent.
+		//   If "false" (the default) we will only collect containers.
+		//   If "true" we will collect containers and processes.
+		//   If "disabled" the agent will be disabled altogether and won't start.
+		_ = log.Warn("process_config.enabled is deprecated, please use process_collection and container_collection instead")
+		enabled := config.Datadog.GetString(key(ns, "enabled"))
+		ok, err := strconv.ParseBool(enabled)
+		if ok {
+			config.Datadog.Set(key(ns, "process_collection", "enabled"), true)
+			config.Datadog.Set(key(ns, "container_collection", "enabled"), true)
+		} else if enabled == "disabled" {
+			config.Datadog.Set(key(ns, "process_collection", "enabled"), false)
+			config.Datadog.Set(key(ns, "container_collection", "enabled"), false)
+		} else if !ok && err == nil {
+			config.Datadog.Set(key(ns, "process_collection", "enabled"), false)
+			config.Datadog.Set(key(ns, "container_collection", "enabled"), true)
+		}
+	}
+
+	if config.Datadog.GetBool(key(ns, "process_collection", "enabled")) {
+		a.EnabledChecks = append(a.EnabledChecks, "process", "process_rt")
```

Comment:
`EnabledChecks` could become a package-wide var to make it easier to remove `AgentConfig`.

Reply:
That's a good idea, I hadn't thought of that before